### ISSUE TITLE
Transition from choice inside a composite state entered for no reason

### ISSUE THREAD

**Issue by MartinOrell** [secure-dev-ops/code-realtime#10]
I have a choice inside a composite state.
The correct transition is made after the choice.
But after the transition is made another transition is also made to the option that shouldn't have been picked

The code is basically this:
```
choice alwaysTrue;
entryPoint->alwaysTrue;
never: alwaysTrue->badExitPoint when `false`;
always: alwaysTrue->SubState when `else`;
```

First the transition is made from alwaysTrue->SubState, then the transition is made to my exitpoint called badExitPoint.

Changing choice into a junction makes the code work the expected way.

Pasting my whole program here:
```
capsule Top{

    [[rt::header_preface]]
    `
        #include <iostream>
    `

    behavior port timer : Timing;

    statemachine{
        initial->CompositeState.entryPoint
        `
            timer.informIn(RTTimespec(1,0));
            std::cout << "initial->CompositeState" << std::endl;
        `;
        state CompositeState{
            entry
            `
                std::cout << "Entered CompositeState" << std::endl;
            `;
            entrypoint entryPoint;
            exitpoint badExitPoint;
            exitpoint goodExitPoint;
            
            choice alwaysTrue;
            
            entryPoint->alwaysTrue
            `
                std::cout << "entryPoint->alwaysTrue" << std::endl;
            `;
            never: alwaysTrue->badExitPoint when `false`
            `
                std::cout << "alwaysTrue->badExitPoint" << std::endl;
            `;
            always: alwaysTrue->SubState when `else`
            `
                std::cout << "alwaysTrue->SubState" << std::endl;
            `;
            state SubState{
                entry
                `
                    std::cout << "Entered SubState" << std::endl;
                `;
            };
            SubState->goodExitPoint on timer.timeout
            `
                std::cout << "SubState->goodExitPoint" << std::endl;
            `;
        };
        CompositeState.badExitPoint->BadState
        `
            std::cout << "badExitPoint->BadState" << std::endl;
        `;
        state BadState{
            entry
            `
                std::cout << "Entered BadState" << std::endl;
                context()->abort();
            `;
        };
        CompositeState.goodExitPoint->GoodState
        `
            std::cout << "goodExitPoint->GoodState" << std::endl;
        `;
        state GoodState{
            entry
            `
                std::cout << "Endered GoodState" << std::endl;
                context()->abort();
            `;
        };
    };
};
```

**Comment by mattias-mohlin** [comment]
@MartinOrell This looks like a bug to me. I was able to reproduce it. There is an unexpected call to the unrelated transition chain function here:
```cpp
case 1 /* RTInitSignal */:                         
  chain5_Initial(  );                         
  chain1(  );                         
  chain6(  ); // This is an unrelated transition chain function that should not be called!
  return ;
```
I have raised a bug in our internal system (reference number RIC-1024) and we have planned it for the 1.0.5 release.

**Comment by alexander-str** [comment]
@MartinOrell The fix is now being considered for 2.0 release in September.

**Comment by astrabykin** [comment]
@MartinOrell this issue is not planned for November release version 2.0.1.

**Comment by astrabykin** [comment]
@MartinOrell This issue has been fixed in version 2.0.1.
